src/prefect/tasks/postgres/postgres.py

```diff
@@ -100,6 +100,106 @@ def run(
+            not provided
+        - query (str, optional): query to execute against database
+            is query string
+        - data (List[tuple], optional): list of values to use in query, must be specified using
+            placeholder
+        - commit (bool, optional): set to True to commit transaction, defaults to false
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        db_name: str,
+        user: str,
+        host: str,
+        port: int = 5432,
+        query: str = None,
+        data: list = None,
+        commit: bool = False,
+        **kwargs
+    ):
+        self.db_name = db_name
+        self.user = user
+        self.host = host
+        self.port = port
+        self.query = query
+        self.data = data
+        self.commit = commit
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("query", "data", "commit")
+    def run(
+        self,
+        query: str = None,
+        data: list = None,
+        commit: bool = False,
+        password: str = None,
+    ):
+        """
+        Task run method. Executes many queries against Postgres database.
+
+        Args:
+            - query (str, optional): query to execute against database
+            - data (List[tuple], optional): list of values to use in query, must be specified using
+                placeholder
+            - commit (bool, optional): set to True to commit transaction, defaults to false
+            - password (str): password used to authenticate; should be provided from a `Secret` task
+
+        Returns:
+            - None
+
+        Raises:
+            - ValueError: if query parameter is None or a blank string
+            - DatabaseError: if exception occurs when executing the query
+        """
+        if not query:
+            raise ValueError("A query string must be provided")
+
+        if not data:
+            raise ValueError("A data list must be provided")
+
+        # connect to database, open cursor
+        # allow psycopg2 to pass through any exceptions raised
+        conn = pg.connect(
+            dbname=self.db_name,
+            user=self.user,
+            password=password,
+            host=self.host,
+            port=self.port,
+        )
+
+        # try to execute query
+        # context manager automatically rolls back failed transactions
+        try:
+            with conn:
+                with conn.cursor() as cursor:
+                    executed = cursor.executemany(query=query, vars_list=data)
+                    if commit:
+                        conn.commit()
+                    else:
+                        conn.rollback()
+
+            conn.close()
+            return executed
+
+        # pass through error, and ensure connection is closed
+        except (Exception, pg.DatabaseError):
+            conn.close()
+            raise
```

Comment:
I'm way too late to the party, but it seems to me that this could be improved with:
```
try:
    with conn, conn.cursor() as cursor:
        executed = cursor.executemany(query=query, vars_list=data)
        if commit:
            conn.commit()
        else:
            conn.rollback()

    return executed

# ensure connection is closed
finally:
    conn.close()
```